build.sbt

```diff
@@ -25,3 +25,28 @@ lazy val core = (projectMatrix in file("core"))
     libraryDependencies ++= Seq(Libraries.uPickle) ++ Libraries.sttpClient ++ Seq(Libraries.scalaTest)
   )
   .settings(commonSettings: _*)
+
+//TODO this should be invoked by compilation process, see #https://github.com/scalameta/mdoc/issues/355
+val compileDocs: TaskKey[Unit] = taskKey[Unit]("Compiles docs module throwing away its output")
+compileDocs := {
+  (docs.jvm(scala2.head) / mdoc).toTask(" --out target/sttp-openai-docs").value
+}
+
+lazy val docs = (projectMatrix in file("generated-docs")) // important: it must not be docs/
```

Comment:
doesn't `generated-docs` need to be in `.gitignore`?

Reply:
ah you are overwriting the out, do you need to specify both `--out` and `mdocOut` then?

Reply:
I guess if we want to check readme's code snippets compile execution only, we could set `mdocOut` to` --out target/sttp-openai-docs` so the output will not affect the repo and be ignored by git. Wdyt?

Reply:
yes putting this into `target` sounds good :) I would also remote the TODOs as I think they don't apply here

Reply:
bump :) this seems to be copied from tapir, along with comments which don't necessarily make sense here